torch/csrc/autograd/variable.cpp

```diff
@@ -64,6 +67,9 @@ void Variable::detach_() {
   if (is_view()) {
     AT_ERROR("Can't detach views in-place. Use detach() instead");
   }
+  // I think the choice here is conservative.  But there is some weirdness going
+  // on: why aren't, e.g., hooks cleared in this case?
+  materialize_autograd_meta();
```

Comment:
Sure

Reply:
My new idea is to just make `materialize_autograd_meta` return an `AutogradMeta*`

Reply:
I think that's slightly worse, because making it part of the get API explicitly forces you to think about it (I don't even need to read the docs really, it's obvious from the parameter name what's going on).

But returning is an improvement.